hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieDropPartitionsTool.java

```diff
@@ -0,0 +1,537 @@
+    sb.append("   --hive-sync-mode ").append(cfg.hiveSyncMode).append(", \n");
+    sb.append("   --hive-sync-ignore-exception ").append(cfg.hiveSyncIgnoreException).append(", \n");
+    sb.append("   --hive-partition-value-extractor-class ").append(cfg.partitionValueExtractorClass).append(", \n");
+    sb.append("   --spark-master ").append(cfg.sparkMaster).append(", \n");
+    sb.append("   --spark-memory ").append(cfg.sparkMemory).append(", \n");
+    sb.append("   --props ").append(cfg.propsFilePath).append(", \n");
+    sb.append("   --hoodie-conf ").append(cfg.configs);
+    sb.append("\n}");
+    return sb.toString();
+  }
+
+  public static void main(String[] args) {
+    final Config cfg = new Config();
+    JCommander cmd = new JCommander(cfg, null, args);
+    if (cfg.help || args.length == 0) {
+      cmd.usage();
+      System.exit(1);
+    }
+    SparkConf sparkConf = UtilHelpers.buildSparkConf("Hoodie-Drop-Table-Partitions", cfg.sparkMaster);
+    sparkConf.set("spark.executor.memory", cfg.sparkMemory);
+    JavaSparkContext jsc = new JavaSparkContext(sparkConf);
+    HoodieDropPartitionsTool tool = new HoodieDropPartitionsTool(jsc, cfg);
+    try {
+      tool.run();
+    } catch (Throwable throwable) {
+      LOG.error("Fail to run deleting table partitions for " + tool.getConfigDetails(), throwable);
+    } finally {
+      jsc.stop();
+    }
+  }
+
+  public void run() {
+    try {
+      if (StringUtils.isNullOrEmpty(cfg.instantTime)) {
+        cfg.instantTime = HoodieActiveTimeline.createNewInstantTime();
+      }
+      LOG.info(getConfigDetails());
+
+      Mode mode = Mode.valueOf(cfg.runningMode.toUpperCase());
+      switch (mode) {
+        case DELETE_PARTITIONS_LAZY:
+          LOG.info(" ****** The Hoodie Drop Partitions Tool is in delete_partition_lazy mode ******");
+          doDeleteTablePartitionsLazy();
+          break;
+        case DELETE_PARTITIONS_EAGER:
+          LOG.info(" ****** The Hoodie Drop Partitions Tool is in delete_partition_eager mode ******");
+          doDeleteTablePartitionsEager();
+          break;
+        case DELETE_PARTITIONS_QUIET:
+          LOG.info(" ****** The Hoodie Drop Partitions Tool is in delete_partition_quiet mode ******");
+          doDeleteTablePartitionsQuiet();
+          break;
+        case DRY_RUN:
+          LOG.info(" ****** The Hoodie Drop Partitions Tool is in dry-run mode ******");
+          dryRun();
+          break;
+        default:
+          LOG.info("Unsupported running mode [" + cfg.runningMode + "], quit the job directly");
+      }
+    } catch (Exception e) {
+      throw new HoodieException("Unable to delete table partitions in " + cfg.basePath, e);
+    }
+  }
+
+  public void doDeleteTablePartitionsLazy() {
+    doDeleteTablePartitions();
+    syncToHiveIfNecessary(false);
+  }
+
+  public void doDeleteTablePartitionsEager() {
+    doDeleteTablePartitions();
+    deleteDataFiles();
```

Comment:
not sure if we can do out of the box deletions w/o going via HoodieWrliteClient. then cleaner may run into issues while trying to clean. Or atleast, list of partition list maintained by metadata table will be out of sync with actual FS. So, I would suggest we should try our best to go via writeClient apis. May be we can improvide deletePartition impl within writeClient to incorporate these add ons (eager delete, lazy deletes, delete partition dir etc).

Reply:
Make Sense! Just thinking, how about designing doDeleteTablePartitionsEager like
1. doDeleteTablePartitions which will call client.deletePartitions 
        a. create a replace commit.
        b. sync metadata table.
        c. schedule and execute an inline clean action immediately.
3. delete empty dir.
4. sync to hive.

The advantage of doing this is that 
1. we do anything in the box via HoodieWrliteClient
2. take care of metadata table syncing.
3. no need to change current writeClient apis.